Monika-After-Story/MonikaModDev
# Background Changing Framework

# Introducing `MASBackground`s:
BG changing framework is now here with this, which introduces `MASBackground` objects

- These objects have a variety of parameters which handle the day/night image to use as the 'room', and optional parameters which are for how the 'room' should look while it's raining, snowing, or overcast. (See the init for a `MASBackground` object for details)
- They also have the parameters to either `hide_masks` (so we don't have weather at all) and `hide_calendar`, which will not display the calendar for the drawing of the spaceroom.
- Backgrounds are also *unlockable* which allows for future expansion.

### Other changes:
- Weather objects now have a `precip_type` associated with them. This allows us to decide what room to use (def, rain, overcast, or snow)
- As a result of allowing potentially different weather to be a case, window masks had to be changed to full size, so any possible future rooms would be supported by our current weather system. (Thank you @Legendkiller21 for helping out in making these, along with @Velius94 and @Orcaramelo for the frost on the snow windows + the night/overcast/space clouds)
- There is also a selector for BGs, but as we do not have any other rooms to use, it's init is commented out. (Feel free to review it for now, but I'd probably save it until we get backgrounds to use it for.)
- Also added the new night bg

## Testing:
- Please make sure everything looks good
- Ensure that spaceroom calls also work right (shouldn't be as bad as last time since not that much changed)
- Ensure that progressive weather adjusts the room correctly
